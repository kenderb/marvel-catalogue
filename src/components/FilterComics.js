import React from 'react';
import PropTypes from 'prop-types';
import { FilterSelect, FilterWrapper } from './styled/lib';
import ComicList from '../containers/ComicList';

const FilterComics = ({ currentFilter, cretorList, onChangeSelect }) => (
  <>
    <FilterWrapper>
      <b>Filter by creator: </b>
      {' '}
      <FilterSelect onChange={e => onChangeSelect(e)}>
        {['All', ...cretorList].map(creator => (
          <option value={creator} key={creator}>{creator}</option>
        ))}
      </FilterSelect>
    </FilterWrapper>
    <ComicList filter={currentFilter} />
  </>
);

FilterComics.propTypes = {
  cretorList: PropTypes.instanceOf(Array),
  onChangeSelect: PropTypes.func,
  currentFilter: PropTypes.string,
};

FilterComics.defaultProps = {
  cretorList: [],
  onChangeSelect: () => (''),
  currentFilter: '',
};

export default FilterComics;
