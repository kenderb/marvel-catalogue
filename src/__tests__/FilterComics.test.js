import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combineReducers from '../reducers';
import FilterComics from '../components/FilterComics';

test('FilterComics renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <FilterComics />
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
