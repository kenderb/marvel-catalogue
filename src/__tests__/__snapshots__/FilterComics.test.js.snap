// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComics renders correctly 1`] = `
Array [
  <div
    className="sc-fubCfw kleonc"
  >
    <b>
      Filter by creator: 
    </b>
     
    <select
      className="sc-pFZIQ jEJBT"
      onChange={[Function]}
    >
      <option
        value="All"
      >
        All
      </option>
    </select>
  </div>,
  <ion-spinner
    name="bubbles"
    style={
      Object {
        "color": "red",
        "margin": "auto",
        "transform": "scale(5)",
      }
    }
  />,
]
`;
